### PR TITLE
feat: improve UI responsiveness

### PR DESCRIPTION
**Pull request by CarmitKl**
Implement UI support for small screens, like laptops and tablets. Smallest value: 600px. On this PR the toasts are overlapping the menu for vertical tablets. Horizontal tablets are without that overlap.

This change is related to this issue:
https://github.com/starkware-libs/starkgate-frontend/issues/26

---
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/starkgate-frontend/95)

### CODE REVIEW

File: src/components/Features/ToastProvider/ToastProvider.js [hunk 1]
```diff
@@ -60,8 +69,9 @@ export const ToastProvider = () => {
   const showAlphaDisclaimerToast = () => {
     toast.success(ALPHA_DISCLAIMER_MSG, {
       id: 'alphaDisclaimer',
-      position: 'bottom-left',
-      icon: '❗'
+      position: breakpoint === 'desktop' ? 'bottom-left' : 'bottom-right',
```
Comment (dan-ziv) [inline src/components/Features/ToastProvider/ToastProvider.js:72]:
```suggestion
      position: Breakpoints[breakpoint] === Breakpoints.DESKTOP ? 'bottom-left' : 'bottom-right',
```

File: src/components/Features/ToastProvider/ToastProvider.module.scss [hunk 2]
```diff
@@ -1,5 +1,22 @@
 @import '../../../index';
 
 .toastProvider {
-  margin-top: #{$--header-height}px;
+  margin-top: #{$--header-height-small}px;
+  @media screen and (min-width: #{$--breakpoint-desktop}px) {
+    margin-top: #{$--header-height}px;
+  }
+
+  & > div > div {
```
Comment (dan-ziv) [inline src/components/Features/ToastProvider/ToastProvider.module.scss:9]: Please use scss sytnax

File: src/enums/Breakpoints.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export const Breakpoints = {
```
Comment (dan-ziv) [inline src/enums/Breakpoints.js:1]: 1. Change all keys to uppercased
2.  Change `Breakpoints` to `Breakpoint`

File: src/components/UI/NetworkMenu/NetworkMenu.module.scss [hunk 4]
```diff
@@ -10,6 +10,12 @@ $box-shadow-color: rgba(0, 0, 0, 0.1);
   height: 100%;
   margin-top: 10px;
   padding: 30px 40px;
+  @media screen and (min-width: #{$--breakpoint-tablet}px) and (max-width: #{$--breakpoint-desktop}px) {
```
Comment (dan-ziv) [inline src/components/UI/NetworkMenu/NetworkMenu.module.scss:13]: Add space

File: src/styles/variables.module.scss [hunk 5]
```diff
@@ -1,7 +1,10 @@
 $--header-height: 80;
-$--footer-height: 50;
+$--header-height-small: 50;
+$--footer-height: 40;
 $--main-offset: $--header-height + $--footer-height + 2;
```
Comment (dan-ziv) [inline src/styles/variables.module.scss:4]: Define 2 as var also

File: src/components/Containers/Header/Header.module.scss [hunk 6]
```diff
@@ -1,14 +1,19 @@
 @import '../../../index';
 
 $height: $--header-height;
+$height-small: $--header-height-small;
 $background-color: $--color-alpha-5;
 $chain-color: $--color-white-1;
 
 .header {
-  height: #{$height}px;
+  @media screen and (min-width: #{$--breakpoint-desktop}px) {
```
Comment (dan-ziv) [inline src/components/Containers/Header/Header.module.scss:9]: Please move it under the styles definitions.

File: src/components/Features/ToastProvider/ToastProvider.js [hunk 7]
```diff
@@ -60,8 +69,9 @@ export const ToastProvider = () => {
   const showAlphaDisclaimerToast = () => {
     toast.success(ALPHA_DISCLAIMER_MSG, {
       id: 'alphaDisclaimer',
-      position: 'bottom-left',
-      icon: '❗'
+      position: breakpoint[breakpoint] === 'DESKTOP' ? 'bottom-left' : 'bottom-right',
```
Comment (dan-ziv) [inline src/components/Features/ToastProvider/ToastProvider.js:72]: What is this?
```
breakpoint[breakpoint] === 'DESKTOP'
```

File: src/components/Features/ToastProvider/ToastProvider.js [hunk 8]
```diff
@@ -134,6 +144,7 @@ export const ToastProvider = () => {
   return (
     <Toaster
       containerClassName={styles.toastProvider}
+      gutter={breakpoint === 'desktop' ? 16 : 8}
```
Comment (dan-ziv) [inline src/components/Features/ToastProvider/ToastProvider.js:147]: Why not to use the enum?

File: src/components/Features/ToastProvider/ToastProvider.js [hunk 9]
```diff
@@ -60,8 +69,9 @@ export const ToastProvider = () => {
   const showAlphaDisclaimerToast = () => {
     toast.success(ALPHA_DISCLAIMER_MSG, {
       id: 'alphaDisclaimer',
-      position: 'bottom-left',
-      icon: '❗'
+      position: breakpoint[breakpoint] === 'DESKTOP' ? 'bottom-left' : 'bottom-right',
+      icon: '❗',
+      className: 'disclaimer ' + breakpoint
```
Comment (dan-ziv) [inline src/components/Features/ToastProvider/ToastProvider.js:74]: Change to:
```
`disclaimer ${breakpoint}`
```

File: src/components/Features/ToastProvider/ToastProvider.module.scss (hunk 2 above)
Comment (dan-ziv) [reply src/components/Features/ToastProvider/ToastProvider.module.scss:10]: Please fix this

File: src/components/UI/Button/Button.js [hunk 10]
```diff
@@ -12,6 +12,7 @@ export const Button = ({
   height,
   icon,
   iconAlign = 'left',
+  buttonClass,
```
Comment (dan-ziv) [inline src/components/UI/Button/Button.js:15]: If you added an option to send class prop, why `buttonClass` as single and not just `classes` as array?

File: src/components/UI/Button/Button.module.scss [hunk 11]
```diff
@@ -28,4 +30,12 @@
   svg {
     margin-right: 10px;
   }
+
+  &[class~='wallet-btn'] {
```
Comment (dan-ziv) [inline src/components/UI/Button/Button.module.scss:34]: Please change to scss syntax

File: src/components/UI/WalletButton/WalletButton.js [hunk 12]
```diff
@@ -12,12 +12,13 @@ export const WalletButton = ({account, logoPath, onClick}) => {
...
   return (
     <Button
+      buttonClass="wallet-btn"
       colorBackground="transparent"
       colorBorder={colorBeta}
       colorText={colorWhite}
       height={40}
       icon={<DynamicIcon path={logoPath} size={WALLET_LOGO_SIZE} />}
-      style={{borderWidth: '2px'}}
+      style={{borderWidth: '2px', padding: '4px 10px 4px 6px'}}
```
Comment (dan-ziv) [inline src/components/UI/WalletButton/WalletButton.js:21]: why not to add this to the class you just sent?

File: src/styles/breakpoints.module.scss [hunk 13]
```diff
@@ -0,0 +1,11 @@
+$--breakpoint-mobile: 0;
+$--breakpoint-tablet: 768;
+$--breakpoint-laptop: 960;
+$--breakpoint-desktop: 1280;
+
+:export {
+  $breakpointMobile: $--breakpoint-mobile;
+  $breakpointTablet: $--breakpoint-tablet;
+  $breakpointLaptop: $--breakpoint-laptop;
+  $breakpointDesktop: $--breakpoint-desktop;
```
Comment (dan-ziv) [inline src/styles/breakpoints.module.scss:10]: No need to export it with a `$`

File: src/styles/variables.module.scss [hunk 14]
```diff
@@ -1,7 +1,11 @@
...
-$--main-offset: $--header-height + $--footer-height + 2;
+$--header-height-small: 50;
+$--footer-height: 40;
+$--main-offset-borders-height: 2;
+$--main-offset: $--header-height + $--footer-height + $--main-offset-borders-height;
+$--main-offset-small: $--header-height-small + $--footer-height + $--main-offset-borders-height;
 
 :export {
   mainOffset: $--main-offset;
+  mainOffsetSmall: $--main-offset-small;
```
Comment (dan-ziv) [inline src/styles/variables.module.scss:10]: What is small? tablet? mobile? please write the appropriate breakpoint.

**Review by dan-ziv — CHANGES_REQUESTED** [review]

**Review by dan-ziv — CHANGES_REQUESTED** [review]
Revert the disclaimer position to left

File: src/styles/variables.module.scss (hunk 14 above)
Comment (CarmitKl) [reply src/styles/variables.module.scss:10]: the '$--header-height' is for width 1280 and above. breakpoint DESKTOP.
and the '$--header-height-small' is for all the rest.

File: src/components/UI/Button/Button.module.scss (hunk 11 above)
Comment (CarmitKl) [reply src/components/UI/Button/Button.module.scss:34]: writing:    &.wallet-btn   doesn't seem to work.

File: src/components/UI/Button/Button.js (hunk 10 above)
Comment (CarmitKl) [reply src/components/UI/Button/Button.js:15]: currently, that's the only class a button gets in the whole project.
also, there's an extended usage of a util function that gets an array of values and returns a string.

File: src/styles/variables.module.scss (hunk 14 above)
Comment (dan-ziv) [reply src/styles/variables.module.scss:10]: So why not to change `$--header-height` to `$--header-height-desktop` and `$--header-height-small` to `$--header-height`???

File: src/components/UI/Button/Button.module.scss (hunk 11 above)
Comment (dan-ziv) [reply src/components/UI/Button/Button.module.scss:34]: 1.It won't work - CSS modules add id to each class at runtime. You should use the `styles` object of the CSS file imported to the component. i.e., `styles.walletBtn`.
2. Why are you adding app-specific class component styles (wallet-btn) to a generic component styles?

File: src/components/Features/ToastProvider/ToastProvider.js [hunk 15]
```diff
@@ -133,7 +147,12 @@ export const ToastProvider = () => {
 
   return (
     <Toaster
+      className="toast"
```
Comment (dan-ziv) [inline src/components/Features/ToastProvider/ToastProvider.js:150]: Where is this call defined?

**Pull request closed without merging**